crates/node/src/cli/mod.rs

```diff
@@ -24,6 +24,60 @@ use mp_chain_config::ChainConfig;
 use std::path::PathBuf;
 use std::sync::Arc;
 
+#[derive(Clone, Debug, clap::Parser)]
+#[clap(
+    group(
+        ArgGroup::new("args-preset")
+            .args(&["warp_update_sender", "warp_update_receiver", "gateway", "rpc"])
+            .multiple(false)
+    )
+)]
+pub struct ArgsPresetParams {
```

Comment:
what's this preset thing and how does it relate to the older presets feature

Reply:
So. Basically I realized some configs were starting to take up a lot of arguments. Initially, warp updates were just set up as a series of predefined arguments (this is still the case for `--warp-update-sender`), and it seemed really tedious and error-prone for the user to have to remember so many options. All this does is that it sets various options in the `RunCmd` struct _after_ parsing has occurred. This has its downsides, most notably the user cannot override args presets this way as they are evaluated _after_ user inputs. The advantage is that this streamlines more complex cli setups with many options and flags (this mostly applies to `--warp-update-sender`).

Reply:
Will be adding this in a doc comment

Reply:
This has been fixed. Added some documentation to `ArgsPresetParams` which detail this, as well as some of the issues and limitation with the current cli setup.